<launch>

<!-- Lidar perception algorithms  -->
  <node name="perception_node" pkg="avt_341" type="avt_341_perception_node" required="true" output="screen">
    <param name="use_elevation" value="true" />
    <param name="slope_threshold" value="0.5" />
    <param name="grid_res" value="2" />
    <param name="grid_llx" value="0.0" />
    <param name="grid_lly" value="-2000.0" />
    <param name="grid_width" value="1100.0" />
    <param name="grid_height" value="2000.0" />
    <param name="warmup_time" value="1.0" />
    <param name="use_registered" value="true" />
  </node>
  
  <node name="vehicle_control_node" pkg="avt_341" type="avt_341_control_node" required="true" output="screen" >
    <param name="vehicle_wheelbase" value="3.3" />
    <param name="vehicle_max_steer_angle_degrees" value="38.7" />
    <param name="steering_coefficient" value="2.5" />
    <param name="vehicle_speed" value="10" />
  </node>
  
  <node name="global_path_node" pkg="avt_341" type="avt_341_global_path_node" required="true" output="screen" >
    <param name="goal_dist" value="3.0" /> 
  </node>
  
  <node name="local_planner_node" pkg="avt_341" type="avt_341_local_planner_node" output="screen" required="true" >
    <param name="path_look_ahead" value="40.0" /> 
    <param name="vehicle_width" value="2.25" />
    <param name="num_paths" value="21" />
    <param name="max_steer_angle" value="0.675" />
    <param name="output_path_step" value="0.5" />
    <param name="path_integration_step" value="0.35" />
    <param name="dilation_factor" value="0" />
    <param name="w_c" value="0.1" />
    <param name="w_s" value="0.4" />
    <param name="w_d" value="0" />
    <param name="w_r" value="0.4" /> 
    <param name="rate" value="10.0" />
    <param name="trim_path" value="false" />
    <param name="display" value="true" />
  </node>
  
</launch>
